.github/workflows/reusable_build_maturin_macos.yml

```diff
@@ -0,0 +1,97 @@
+name: reusable_build_maturin_macos
+
+on:
+  workflow_dispatch:
+    inputs:
+      py_interface_folder:
+        description: "Location of the py interface folder"
+        required: true
+        type: string
+      deploy:
+        description: "Determines if build python wheels should be deployed to PyPi"
+        required: false
+        default: false
+        type: boolean
+      universal2:
+        description: "Build python wheels with universal2 flag"
+        required: false
+        default: true
+        type: boolean
+  workflow_call:
+    inputs:
+      py_interface_folder:
+        description: "Location of the py interface folder"
+        required: true
+        type: string
+      deploy:
+        description: "Determines if build python wheels should be deployed to PyPi"
+        required: false
+        default: false
+        type: boolean
+      universal2:
+        description: "Build python wheels with universal2 flag"
+        required: false
+        default: true
+        type: boolean
+
+jobs:
+
+  build_maturin_builds_macos:
+    name: maturin_build-macos-${{ matrix.python.interpreter }}
+    runs-on: 'macOS-latest'
+    strategy:
+      matrix:
+        python: [
+            {py: '3.10', interpreter: "python3.10"},
+            {py: '3.11', interpreter: "python3.11"}
+        ]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: "aarch64-apple-darwin"
+          default: true
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python.py }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip pytest numpy twine
+          python -m pip install maturin
+      - name: macos wheels
+        if: ${{ inputs.universal2 == false || matrix.python.py == '3.7'}}
```

Comment:
Only python 3.10 and 3.11 are specified in the matrix, do we need the extra part of the if statement?